File: Embarcadero/XE8/CPP/Forms/LandscapeForm.cpp
//---------------------------------------------------------------------------

// This software is Copyright (c) 2013 Embarcadero Technologies, Inc.
// You may only use this software if you are an authorized licensee
// of Delphi, C++Builder or RAD Studio (Embarcadero Products).
// This software is considered a Redistributable as defined under
// the software license agreement that comes with the Embarcadero Products
// and is subject to that software license agreement.

//---------------------------------------------------------------------------

#include <fmx.h>
#pragma hdrstop

#include "LandscapeForm.h"
#include "PortraitForm.h"
//---------------------------------------------------------------------------
#pragma package(smart_init)
#pragma resource "*.fmx"
#pragma resource ("*.LgXhdpiPh.fmx", _PLAT_ANDROID)

TLSForm *LSForm;
//---------------------------------------------------------------------------
__fastcall TLSForm::TLSForm(TComponent* Owner)
	: TForm(Owner)
{
}
//---------------------------------------------------------------------------
void __fastcall TLSForm::FormCreate(TObject *Sender)
{
	FCreated = True;
}
//---------------------------------------------------------------------------

void __fastcall TLSForm::FormResize(TObject *Sender)
{
	if((Height > Width) && (Visible) && (PForm != NULL)) {
        PForm->Show();
    }
}
//---------------------------------------------------------------------------

File: Embarcadero/XE7/ObjectPascal/Forms/LandscapeForm.pas
//---------------------------------------------------------------------------

// This software is Copyright (c) 2013 Embarcadero Technologies, Inc.
// You may only use this software if you are an authorized licensee
// of Delphi, C++Builder or RAD Studio (Embarcadero Products).
// This software is considered a Redistributable as defined under
// the software license agreement that comes with the Embarcadero Products
// and is subject to that software license agreement.

//---------------------------------------------------------------------------

unit LandscapeForm;

interface

uses
  System.SysUtils, System.Types, System.UITypes, System.Classes, System.Variants,
  FMX.Types, FMX.Controls, FMX.Forms, FMX.Dialogs, FMX.StdCtrls, FMX.Objects, FMX.MobilePreview;

type
  TLSForm = class(TForm)
    ToolBar1: TToolBar;
    Label2: TLabel;
    Image1: TImage;
    procedure FormResize(Sender: TObject);
    procedure FormCreate(Sender: TObject);
    procedure FormKeyUp(Sender: TObject; var Key: Word; var KeyChar: Char;
      Shift: TShiftState);
  private
    FCreated: Boolean;
  public
    { Public declarations }
  end;

var
  LSForm: TLSForm;

implementation

uses
 PortraitForm;

{$R *.fmx}
{$R *.LgXhdpiPh.fmx ANDROID}

procedure TLSForm.FormCreate(Sender: TObject);
begin
  FCreated := True;
end;

procedure TLSForm.FormKeyUp(Sender: TObject; var Key: Word; var KeyChar: Char;
  Shift: TShiftState);
begin
{$ifdef ANDROID}
  if Key = vkHardwareBack then
    Key := 0; // avoid the default back action.
{$endif}
end;

procedure TLSForm.FormResize(Sender: TObject);
begin
  if (Height > Width) and (Visible) and Assigned(PForm) then
    PForm.Show;
end;

end.
